### PR TITLE
Remove MPI from python bindings

### PR DESCRIPTION
**Pull request by BenjaminRueth**
The way how the python bindings are using MPI makes the code more verbose, complicated compilation and leads to a confusing dependency on mpi4py (#8). As far as I know we currently do not have a use-case, where MPI is really required for the python bindings. We already made good experiences entirely removing the MPI from the python bindings (https://precice.discourse.group/t/installation-of-python-bindings-on-cray/149/4?u=benjaminrueth).

Therefore, I would suggest to entirely remove MPI from the python bindings. If anybody knows a use-case, where we cannot remove MPI from the python bindings as suggested in this PR, please feel free to comment.

### CODE REVIEW

**Comment by uekerman** [comment]
> As far as I know we currently do not have a use-case, where MPI is really required for the python bindings.

Testing a simple parallel solver dummy should be fairly easy.

**Comment by BenjaminRueth** [comment]
> > As far as I know we currently do not have a use-case, where MPI is really required for the python bindings.
> 
> Testing a simple parallel solver dummy should be fairly easy.

:+1: I will add a test for this.

File: solverdummy/README.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ You can test the dummy solver by coupling two instances with each other. Open tw
  * `python3 solverdummy.py precice-config.xml SolverOne MeshOne`
  * `python3 solverdummy.py precice-config.xml SolverTwo MeshTwo`
 
+## parallel version
+
+For the parallel verison of the solverdummy, you have to run add the command `mpirun -n <N>` to the  commands from above.
```
Comment (MakisH) [inline solverdummy/README.md]:
```suggestion
For the parallel verison of the solverdummy, you have to add the command `mpirun -n <N>` to the  commands from above.
```

**Comment by BenjaminRueth** [comment]
> 
> 
> > > As far as I know we currently do not have a use-case, where MPI is really required for the python bindings.
> > 
> > 
> > Testing a simple parallel solver dummy should be fairly easy.
> 
> 👍 I will add a test for this.

Test works. Also without MPI.

File: setup.py [hunk 2]
```diff
@@ -21,53 +21,13 @@
...
-    else:  # if PRECICE_MPI_IMPLEMENTATION is not mpich or openmpi quit.
-        raise Exception("unknown/no mpi found using compiler %s. Could not build PySolverInterface." % mpi_compiler_wrapper)
-
-    return mpi_compile_args, mpi_link_args
-
-
-def get_extensions(mpi_compiler_wrapper, is_test):
+def get_extensions(is_test):
     compile_args = []
     link_args = []
```
Comment (fsimonis) [inline setup.py]: This list can now contain only a single argument. Maybe there is a more elegant way of writing this?

Comment (BenjaminRueth) [reply setup.py]: The one-element-list is indeed a bit strange, but I currently do not really see a nicer solution. Note that we also want to be able to provide an empty list, if we are running the tests. I would keep it as it is.

**Pull request merged**